20928 Add details about HTMLMetaElement API usage with media (#21511)
mdn/content: files/en-us/web/api/htmlmetaelement/index.md

{{ APIRef("HTML DOM") }}

> The **`HTMLMetaElement`** interface contains descriptive metadata about a document provided in HTML as [`<meta>` elements](/en-US/docs/Web/HTML/Element/meta).
> It inherits all of the properties and methods described in the {{domxref("HTMLElement")}} interface.

Reviewer: ```suggestion
The **`HTMLMetaElement`** interface contains descriptive metadata about a document provided in HTML as [`<meta>`](/en-US/docs/Web/HTML/Element/meta) elements.
This interface inherits all of the properties and methods described in the {{domxref("HTMLElement")}} interface.
```

{{InheritanceDiagram}}

## Instance properties

_Inherits properties from its parent, {{domxref("HTMLElement")}}._

- {{domxref("HTMLMetaElement.charset")}}

>   - : The character encoding for a HTML document.

Reviewer 2: Not 100% sure if the following is true, but if it is, we should include it:
```suggestion
  - : The character encoding for a HTML document and the text-based documents it imports.
```

I read somewhere recently that you don't have to add encoding the linked or imported stylesheets, but was not able to confirm it. If this is true, we should let devs know. If it's not true, let me know ;)

Author: I found

https://www.w3.org/International/questions/qa-html-encoding-declarations#charset

>The HTML 4.01 specification describes a charset attribute that can be added to the link element to indicate the encoding of the document you are linking to. However, the use of this attribute on a link element is currently [obsoleted](http://www.w3.org/TR/html5/obsolete.html#attr-link-charset) by the HTML5 specification, so you shouldn't use it.

It shouldn't be done in `<link>` and it's discouraged to use anything except UTF-8 by the looks of it. In the rare cases you need different encodings, they recommend headers and in-file declaration: https://www.w3.org/International/questions/qa-html-encoding-declarations#httphead

- {{domxref("HTMLMetaElement.content")}}
  - : The 'value' part of name/value pairs of document metadata.
- {{domxref("HTMLMetaElement.httpEquiv")}}

>   - : The name of an HTTP response header to define for a document.

Reviewer 2: ```suggestion
  - : The name of the pragma directive, the HTTP response header, to define for a document.
```

- {{domxref("HTMLMetaElement.media")}}

>   - : The media context for a metadata property.

Reviewer 2: ```suggestion
  - : The media context for a `theme-color` metadata property.
```

- {{domxref("HTMLMetaElement.name")}}

>   - : The 'name' part of name/value pairs defining metadata of a document.

Reviewer 2: ```suggestion
  - : The 'name' part of name/value pairs defining named metadata of a document.
```

> - {{domxref("HTMLMetaElement.scheme")}} {{deprecated_inline}}

Reviewer 2: should we even include this and the scheme page? If it was supported, yes. if not, in don't this we should.

Author: It's supported still, but it didn't make it from HTML4 -> HTML5. This works, though:

```js
// if a page has <meta name="identifier" content="1580081754" scheme="ISBN" />
let meta = document.querySelector("meta[name='identifier']");
console.log(meta.scheme);
// ISBN
``` 

Took some digging, but here's the ref -> https://www.w3.org/TR/html5-diff/#obsolete-attributes:

>[scheme](http://www.w3.org/TR/html5/single-page.html#attr-meta-scheme) attribute on [meta](http://www.w3.org/TR/html5/single-page.html#the-meta-element).

Author: I'm leaning towards leaving it in there, what do you think?

> - {{domxref("HTMLMetaElement.content")}}
>   - : The 'value' part of name/value pairs of document metadata.
> - {{domxref("HTMLMetaElement.httpEquiv")}}
>   - : The name of the pragma directive, the HTTP response header, to define for a document.
> - {{domxref("HTMLMetaElement.media")}}
>   - : The media context for a `theme-color` metadata property.
> - {{domxref("HTMLMetaElement.name")}}
>   - : The 'name' part of name/value pairs defining named metadata of a document.
> - {{domxref("HTMLMetaElement.scheme")}} {{deprecated_inline}}

Reviewer: ```suggestion
- {{domxref("HTMLMetaElement.content")}}
  - : The 'value' part of the name-value pairs of the document metadata.
- {{domxref("HTMLMetaElement.httpEquiv")}}
  - : The name of the pragma directive, the HTTP response header, for a document.
- {{domxref("HTMLMetaElement.media")}}
  - : The media context for a `theme-color` metadata property.
- {{domxref("HTMLMetaElement.name")}}
  - : The 'name' part of the name-value pairs defining the named metadata of a document.
- {{domxref("HTMLMetaElement.scheme")}} {{deprecated_inline}}
```

  - : Defines the scheme of the value in the {{domxref("HTMLMetaElement.content")}} attribute.
    This is deprecated and should not be used on new web pages.

## Instance methods

_No specific method; inherits methods from its parent, {{domxref("HTMLElement")}}._

## Examples

> The following teo examples show a general approach to using the `HTMLMetaElement` interface.

Reviewer 2: ```suggestion
The following two examples show a general approach to using the `HTMLMetaElement` interface.
```

> For specific examples, see the pages for the individual properties as described in the [Instance properties](#instance_properties) above.

Reviewer: ```suggestion
For specific examples, see the pages for the individual properties as described in the [Instance properties](#instance_properties) section above.
```

> ### Setting page description metadata

Reviewer: ```suggestion
### Setting the page description metadata
```

> Creating a new meta element with `description` as a name and setting a `content` value:

Reviewer 2: ```suggestion
Creating a new `<meta>` element with `description` as a name and setting a `content` value:
```

> Creating a new `<meta>` element with `description` as a name and setting a `content` value:

Reviewer: ```suggestion
The following example shows how to create a new `<meta>` element with `description` as a name and setting a `content` value:
```

```js
let meta = document.createElement("meta");
meta.name = "description";
meta.content =
  "The <meta> element can be used to provide document metadata in terms of name-value pairs, with the name attribute giving the metadata name, and the content attribute giving the value.";
document.head.appendChild(meta);
```

> ### Setting the viewport

Reviewer 2: ```suggestion
### Setting the viewport metadata
```

> 

Reviewer: a description for the example would be nice
